packages/language-core/lib/codegen/script/template.ts

```diff
@@ -257,3 +261,15 @@ export function getTemplateUsageVars(options: ScriptCodegenOptions, ctx: ScriptC
 
 	return usageVars;
 }
+
+function getRefsType(options: ScriptCodegenOptions, scriptSetupRanges: ScriptSetupRanges) {
```

Comment:
Can we use the generator here?

Reply:
I tried. But `const exps = []` is a string[] type, Can you try it?

Reply:
I found that `generateTemplateCtx` also needs to be converted into a pure generator-based form. In this case, its readability doesn't seem to differ much from using array operations :)